src/Infrastructure.VS/ThreadHandling.cs

```diff
@@ -42,22 +43,28 @@ public class ThreadHandling : IThreadHandling
 
         public async Task<T> RunOnBackgroundThread<T>(Func<Task<T>> asyncMethod)
         {
+            CodeMarkers.Instance.Threading_RunOnBackgroundThread_Start();
+
             var startedOnUIThread = ThreadHelper.CheckAccess();
 
             return await ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                 {
                     if (startedOnUIThread)
                     {
+                        CodeMarkers.Instance.Threading_RunOnBackgroundThread_SwitchingToBackground();
                         await SwitchToBackgroundThread();
                     }
 
+                    CodeMarkers.Instance.Threading_RunOnBackgroundThread_ExecutingCall();
                     var result = await asyncMethod();
 
                     if (startedOnUIThread)
                     {
+                        CodeMarkers.Instance.Threading_RunOnBackgroundThread_SwitchingToUI();
```

Comment:
Question about Switching ToBackground/ToMain: they will be logged on the other thread; is that a problem for us when investigating?
i.e. we will see "switching to main thread" but the thread id will be background, and vice-versa.

Reply:
I think we'll have enough information because the execute is logged and we'll see what thread that is running on I that will tell us that it was successful.

If not, I'll change it.